core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -544,6 +544,7 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
         case Some(billingAccount) if billingAccount.firecloudHasAccess => Future.successful(billingAccount)
         case None => Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, createForbiddenErrorMessage("You", billingAccountName))))
         case Some(billingAccount) if !billingAccount.firecloudHasAccess => Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, createForbiddenErrorMessage(gcsDAO.billingEmail, billingAccountName))))
+        case Some(billingAccount) => Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, s"Unexpected billing account ${billingAccount}")))
```

Comment:
Thanks for this. I feel this extra pattern match is unnecessary as either firecloud has access to the billing account or it does not. @gpolumbo-broad is that right?
```suggestion
        case Some(billingAccount) => if (billingAccount.firecloudHasAccess)
          Future.successful(billingAccount) else
          Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, createForbiddenErrorMessage(gcsDAO.billingEmail, billingAccountName))))
        case None => Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, createForbiddenErrorMessage("You", billingAccountName))))
```

Reply:
the pattern match has to be exhaustive..updated it to 
```
        case Some(billingAccount) =>
          if (billingAccount.firecloudHasAccess)
            Future.successful(billingAccount)
          else
            Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, createForbiddenErrorMessage(gcsDAO.billingEmail, billingAccountName))))
```

Reply:
Sorry - that was what I was trying to suggest! Thanks!